mypy/checkexpr.py

```diff
@@ -709,6 +709,8 @@ def check_arg(self, caller_type: Type, original_caller_type: Type,
         elif isinstance(caller_type, DeletedType):
             messages.deleted_as_rvalue(caller_type, context)
         elif not is_subtype(caller_type, callee_type):
+            if self.chk.should_suppress_optional_error([caller_type, callee_type]):
```

Comment:
Shouldn't this just use `[caller_type]`? If `callee_type` contains `None` somewhere the error shouldn't be caused by `--strict-optional`?